### ISSUE TITLE
UX: Extra entries in `curie_map` of written TSV

### ISSUE THREAD

**Issue by joeflack4** [mapping-commons/sssom-py#514]
## Overview
When I write SSSOM to TSV, I'd like it to only include entries in `curie_map` where the prefixes are actually used in 1 or more places in the mapping set. However, extra entries are appearing.

## Case 1: When passing a `Converter` and `metadata`
I expected 4 entries, but got 10.
I also mentioned this in: #513

## Case 2: When not passing `metadata`, but no `Converter`
I expected 4 entries, but got 1,547.

## Additional details
FYI:
- Here's the file I'm using to instantiate my `metadata`: [icd11.sssom-metadata.yml.zip](https://github.com/mapping-commons/sssom-py/files/14895252/icd11.sssom-metadata.yml.zip)
- This [is the PR](https://github.com/monarch-initiative/icd11/pull/12) where I'm creating this mapping set.

Results, based on various means:
1. n=4. This is the amount of entries I want, and it is the amount that I get when using some custom code I wrote. [ordo-icd11.sssom - joes ad hoc way.tsv.zip](https://github.com/mapping-commons/sssom-py/files/14895242/ordo-icd11.sssom.-.joes.ad.hoc.way.tsv.zip)
2. n=10. This is the amount of entries I get when first instantiating a `Converter` and passing that to `MappingSetDataFrame`. [ordo-icd11.sssom - with converter.tsv.zip](https://github.com/mapping-commons/sssom-py/files/14895244/ordo-icd11.sssom.-.with.converter.tsv.zip)
3. n=1,547. This is the amount of entries I get when instantiating a `MappingSetDataFrame` passing in my `metadata` but no `Converter`. [ordo-icd11.sssom - no converter.tsv.zip](https://github.com/mapping-commons/sssom-py/files/14895243/ordo-icd11.sssom.-.no.converter.tsv.zip)

**Comment by matentzn** [comment]
1. owl, sssom, and oboInOwl, rdf, rdfs, and semapv are all built in prefixes and are therefore added by default (especially `sssom` and `semapv` which are nearly always needed.
2. There is a method on `MappingSetPrefixMap` called `clean_prefix_map()` which removes all prefixes from the curie map that are not used. Try to see if it gets rid of some of the buildin ones?

**Comment by joeflack4** [comment]
For (2), right--I forgot to do `msdf.clean_prefix_map()`. This reveals 2 things: (i) I'm a dumb dumb, (ii) the UX around this isn't dumb dumb proof.

Or maybe (ii) is wrong? Do we really want the default behavior to be, that when no `converter` is passed that we include all of these 1,547 entries? Not to mention the related sub-issues of #513:  (2) Incorrect `curie_map` (it leaves out prefixes that are in `metadata`), and (3)  UX: Should automatically instantiate Converter.

My preference would be that `clean_prefix_map()` should be automatic, and if we want to have a parameter that adds tons of namespaces, we can add that.

**Comment by matentzn** [comment]
> Do we really want the default behavior to be, that when no converter is passed that we include all of these 1,547 entries?

I think you are right. Can you update the OC to request that, if the massive OBO context was used to infer prefixes, we should automatically call `clean_prefix_map()`?

> Incorrect curie_map (it leaves out prefixes that are in metadata),

If this is true is a bug, add as action item to OC.

> Should automatically instantiate Converter.

This must already be the case..

**Comment by joeflack4** [comment]
(i) Done! (ii) Done!

(iii) It probably is, but let me rephrase: **Should _correctly_ instantiate `Converter`**.
This is related to #513 so let me add what I mean there.
